src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs

```diff
@@ -116,6 +117,16 @@ private async Task RenderComponentCore(HttpContext context)
             }
         }
 
+        if (!quiesceTask.IsCompleted)
+        {
+            // An incomplete QuiescenceTask indicates there may be streaming rendering updates.
+            // Disable all response buffering and compression on IIS like SignalR's ServerSentEventsServerTransport does.
+            var bufferingFeature = context.Features.GetRequiredFeature<IHttpResponseBodyFeature>();
+            bufferingFeature.DisableBuffering();
+
+            context.Response.Headers.ContentEncoding = "identity";
```

Comment:
Will this work if someone enables compression in Kestrel? Does it need to be configurable in that scenario?

Reply:
The only way to enable compression in Kestrel is to use the middleware which respects the `DisableBuffering()` call above.

https://github.com/dotnet/aspnetcore/blob/e1ebf00fbe63aa372051ef1c184895f169ca49f5/src/Middleware/ResponseCompression/src/ResponseCompressionBody.cs#L270-L283